src/erlfmt_cli.erl

```diff
@@ -83,18 +93,48 @@ do_unprotected(Opts, Name) ->
             erlang:halt(2)
     end.
 
+-dialyzer({no_improper_lists, [read_stdin/1]}).
+
+read_stdin(Data) ->
+    case io:get_chars(standard_io, "", 4096) of
+        MoreData when is_binary(MoreData) -> read_stdin([Data | MoreData]);
+        eof -> {ok, Data};
+        {error, Reason} -> throw({error, Reason})
```

Comment:
Do we handle throws nicely in this context? We handle them in `erlfmt`, but I'm not sure here

Reply:
do calls and catch throws from do_unprotected which calls format_file, so I think we are fine.
write_formatted also contained throws which we didn't try to catch here.